src/jquery/ajax.js

```diff
@@ -21,4 +23,26 @@ jQuery.ajax = function( ) {
 	return jQXHR;
 };
 
+// Only trigger the logic in jQuery <4 as the JSON-to-JSONP auto-promotion
+// behavior is gone in jQuery 4.0 and as it has security implications, we don't
+// want to restore the legacy behavior.
+if ( !jQueryVersionSince( "4.0.0" ) ) {
+
+	// Register this prefilter before the jQuery one. Otherwise, a promoted
+	// request is transformed into one with the script dataType and we can't
+	// catch it anymore.
+	jQuery.ajaxPrefilter( "+json", function( s ) {
```

Comment:
The special `+` handling is not documented at https://api.jquery.com/jQuery.ajaxPrefilter/, I found it by reading source. 🙈